New blog post: 1.8.0 release announcement (#34)
cocotb/cocotb-web: _posts/2023-06-15-v1.8.0.md

---
layout: post
title:  cocotb 1.8 is out and makes your verification journey even more enjoyable
author: Philipp Wagner
---

The cocotb project is proud to announce the immediate release of cocotb version 1.8.0.

> This release focuses bug fixes and reliability improvements, but some notable additions as well.

Reviewer: ```suggestion
This release focuses bug fixes and reliability improvements, with some notable additions as well.
```

<!--more-->

> One highly visible change is the expanded assert rewrite.

Reviewer 2: ```suggestion
One highly visible change is the expanded assertion rewriting.
```

When an `assert` statement fails, cocotb displays (with the help of pytest) the actual values that made the assertion fail, not only the variable names.

> This mechanism is called "assert rewriting," and now enabled not only for the test module (the one passed with the `MODULE` environment variable), but also for code in other modules (e.g., helpers, bus adapters, etc.).

Reviewer 2: ```suggestion
This mechanism is called "assertion rewriting," and is now enabled not only for the test module (the one passed with the `MODULE` environment variable), but also for code in other modules (e.g., helpers, bus adapters, etc.).
```

A notable addition is the cocotb runner, a Python-based replacement for the Makefile-based build system to trigger a simulation run.

> The runner *experimental* at this point and focuses on cocotb's own need to run tests.

Reviewer 2: ```suggestion
The runner is *experimental* at this point and focuses on cocotb's own need to run tests.
```

* [@ddribin](https://github.com/ddribin) made their first contribution in [PR 3294](https://github.com/cocotb/cocotb/pull/3294)
* [@Forty-Bot](https://github.com/Forty-Bot) made their first contribution in [PR 3312](https://github.com/cocotb/cocotb/pull/3312)
* [@tj-scherer](https://github.com/tj-scherer) made their first contribution in [PR 3321](https://github.com/cocotb/cocotb/pull/3321)
* [@Ingrimmel](https://github.com/Ingrimmel) made their first contribution in [PR 3329](https://github.com/cocotb/cocotb/pull/3329)
* [@mczyz-antmicro](https://github.com/mczyz-antmicro) made their first contribution in [PR 3316](https://github.com/cocotb/cocotb/pull/3316)

These numbers are impressive and showcase the power of the free and open source collaboration model.
To be able to sustain this amount of change and the high-quality review process behind it we are happy to have an active group of maintainers caring for cocotb and its users.
Thank you, and thanks to the whole cocotb community for coming together to create this unique piece of software.

And now, go and enjoy the best release of cocotb so far!

> If you have questions or issues with this release head over to the [issue tracker](https://github.com/cocotb/cocotb/issues).

Reviewer: Or open a discussion.
